diff --git a/CMakeLists.txt b/CMakeLists.txt
index db9e371..963793b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -844,6 +844,7 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDOWS)
 
   install(DIRECTORY include/rocksdb COMPONENT devel DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 
+  if (NOT BUILD_SHARED_LIBS)
   install(
     TARGETS ${ROCKSDB_STATIC_LIB}
     EXPORT RocksDBTargets
@@ -851,7 +852,7 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDOWS)
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
   )
-
+  else()
   install(
     TARGETS ${ROCKSDB_SHARED_LIB}
     EXPORT RocksDBTargets
@@ -861,7 +862,9 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDOWS)
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
   )
+  endif()
 
+  if (NOT)
   install(
     EXPORT RocksDBTargets
     COMPONENT devel
@@ -876,6 +879,7 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDOWS)
     COMPONENT devel
     DESTINATION ${package_config_destination}
   )
+  endif()
 endif()
 
 option(WITH_TESTS "build with tests" ON)
diff --git a/cmake/modules/ReadVersion.cmake b/cmake/modules/ReadVersion.cmake
index ae356d9..ebfd7d6 100644
--- a/cmake/modules/ReadVersion.cmake
+++ b/cmake/modules/ReadVersion.cmake
@@ -1,7 +1,7 @@
 # Read rocksdb version from version.h header file.
 
 function(get_rocksdb_version version_var)
-  file(READ "${CMAKE_SOURCE_DIR}/include/rocksdb/version.h" version_header_file)
+  file(READ "${CMAKE_CURRENT_SOURCE_DIR}/include/rocksdb/version.h" version_header_file)
   foreach(component MAJOR MINOR PATCH)
     string(REGEX MATCH "#define ROCKSDB_${component} ([0-9]+)" _ ${version_header_file})
     set(ROCKSDB_VERSION_${component} ${CMAKE_MATCH_1})
